typings/objection/index.d.ts

```diff
@@ -1024,4 +1063,91 @@ declare namespace Objection {
 
     QueryBuilderType: QueryBuilder<this, this[]>;
   }
+
+  /**
+   * Overloading is required here until the following issues (at least) are resolved:
+   *
+   * - https://github.com/microsoft/TypeScript/issues/1360
+   * - https://github.com/Microsoft/TypeScript/issues/5453
+   *
+   * @tutorial https://vincit.github.io/objection.js/guide/transactions.html#creating-a-transaction
+   */
+  export interface transaction<T> {
+    start(knexOrModel: knex | ModelClassSpecifier): Promise<Transaction>;
```

Comment:
`ModelClassSpecifier` is not allowed here. Only a model class is. You can copy paste most of this from the old typings.

Reply:
Fixed. One question I had was around the use of `V` - I can't tell what it is by the name but I assume it represents the type of the return value. Would you mind if I renamed it to something more meaningful? Maybe `ReturnValue`?

Reply:
By all means. Descriptive names are always better. I only use short type names `M` and `QB` so that they don't bloat the code.